// backpack-ui/dist/components/cardShelfVideo/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _cardShelf = require("../cardShelf");

var _cardVideo = require("../cardVideo");

var _cardVideo2 = _interopRequireDefault(_cardVideo);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var scopedStyles = {
  ".Card": {
    flex: "0 0 auto",
    width: "100%"
  },

  ".Card + .Card": {
    marginLeft: "30px"
  }
};

var CardShelfVideo = function CardShelfVideo(_ref) {
  var children = _ref.children,
      heading = _ref.heading,
      href = _ref.href,
      style = _ref.style;
  return _react2.default.createElement(
    _cardShelf.CardShelf,
    {
      className: "CardShelf--video",
      style: style
    },
    _react2.default.createElement(_radium.Style, {
      scopeSelector: ".CardShelf--video",
      rules: scopedStyles
    }),
    heading && _react2.default.createElement(_cardShelf.CardShelfHeader, {
      heading: heading,
      href: href
    }),
    _react2.default.createElement(
      _cardShelf.CardShelfContent,
      null,
      children
    )
  );
};

CardShelfVideo.propTypes = {
  children: function children(props, propName, componentName) {
    var prop = props[propName];
    var error = null;

    _react2.default.Children.forEach(prop, function (child) {
      if (child.type !== _cardVideo2.default) {
        error = new Error(componentName + " children should be of type \"CardVideo\".");
      }
    });

    return error;
  },
  heading: _propTypes2.default.string,
  href: _propTypes2.default.string,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(CardShelfVideo);
