// @lonelyplanet/open-planet-node/src/utils/imgix.ts
const providers = {
  bdc: {
    baseUrl: "aff.bstatic.com",
    domain: "https://lpbookingcom.imgix.net",
    regex: /http:\/\/aff.bstatic.com/,
  },
  gadventures: {
    baseUrl: "media.gadventures.com",
    domain: "https://lpgadventures.imgix.net",
    regex: /https:\/\/media.gadventures.com/,
  },
  hw: {
    baseUrl: "ucd.hwstatic.com",
    domain: "https://lphostelworld.imgix.net",
    regex: /http:\/\/(.*)hwstatic.com/,
  },
  media: {
    baseUrl: "media.lonelyplanet.com",
    domain: "https://lonelyplanetimages.imgix.net",
    regex: /https:\/\/media.lonelyplanet.com/,
  },
  viator: {
    baseUrl: "cache-graphicslib.viator.com",
    domain: "https://lpviator.imgix.net",
    regex: /http:\/\/cache-graphicslib.viator.com/,
  },
};

export default function imgix(url, provider = "media") {
  if (Object.keys(providers).indexOf(provider) === -1) { provider = "media"; }
  if (!providers[provider].regex.test(url)) {
    return url;
  }

  return url.replace(providers[provider].regex, providers[provider].domain);
}
