// @lonelyplanet/open-planet-node/src/resources/tag/index.ts
import { injectable } from "inversify";
import {
  IJsonApiResponse,
  IJsonApiResource,
  IJsonApiRelationship,
  IJsonApiRelationships,
} from "../../interfaces";

export const tagType = "tag";
export const tagAssociationType = "tag-association";

export interface ITagAttributes {
  name: string;
}

export interface ITagRelationships extends IJsonApiRelationships {
  "tag-associations": IJsonApiRelationship;
}

export interface ITagResource extends IJsonApiResource<ITagAttributes> {}
export interface ITagResponse extends IJsonApiResponse<ITagResource> {}

export interface ITopicAttributes {
  label: string;
}
export interface ITopicResource extends IJsonApiResource<ITopicAttributes> {}
export interface ITopicResponse extends IJsonApiResponse<ITopicResource> {}

export interface ITagAssociationAttributes {}
export interface ITagAssociationResource extends IJsonApiResource<ITagAssociationAttributes> {}
export interface ITagAssociationResponse extends IJsonApiResponse<ITagAssociationResource> {}

export interface ITag {
  id?: string;
  name?: string;
  targetId?: string;
  targetType?: string;
  associationId?: string;
}

export default class Tag implements ITag {
  id?: string;
  name?: string;
  targetId?: string;
  targetType?: string;
  associationId?: string;
}

export interface ITopic {
  id?: string;
  label?: string;
}

export class Topic implements ITopic {
  id?: string;
  label?: string;
}
