// @lonelyplanet/open-planet-node/dist/openPlanetTypes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.article = "article";
exports.narrative = "narrative";
exports.news = "news";
exports.shop = "shop";
exports.topic = "topic";
exports.video = "video";
exports.thorntree = "thorntree";
exports.thorntreeTopic = "thorntree-topic";
exports.thorntreeForum = "thorntree-forum";
var availability_1 = require("./resources/availability");
exports.availability = availability_1.availabilityType;
var blogCategory_1 = require("./resources/blogCategory");
exports.blogCategory = blogCategory_1.blogCategoryType;
var blogPost_1 = require("./resources/blogPost");
exports.blogPost = blogPost_1.blogPostType;
var bookmark_1 = require("./resources/bookmark");
exports.bookmarkEntry = bookmark_1.bookmarkEntryType;
var bookmarkList_1 = require("./resources/bookmarkList");
exports.bookmarkList = bookmarkList_1.bookmarkListType;
var bookmarkPopularity_1 = require("./resources/bookmarkPopularity");
exports.bookmarkPopularity = bookmarkPopularity_1.bookmarkPopularityType;
var image_1 = require("./resources/image");
exports.image = image_1.imageType;
var lodging_1 = require("./resources/lodging");
exports.lodging = lodging_1.lodgingType;
var partnerActivity_1 = require("./resources/partnerActivity");
exports.partnerActivity = partnerActivity_1.partnerActivityType;
var place_1 = require("./resources/place");
exports.place = place_1.placeType;
var poi_1 = require("./resources/poi");
exports.poi = poi_1.poiType;
var search_1 = require("./resources/search");
exports.searchResult = search_1.searchType;
var tag_1 = require("./resources/tag");
exports.tag = tag_1.tagType;
var user_1 = require("./resources/user");
exports.user = user_1.userType;
var association_1 = require("./resources/association");
exports.association = association_1.associationType;
var associationEntry_1 = require("./resources/associationEntry");
exports.AssociationEntry = associationEntry_1.associationEntryType;
var blacklistItem_1 = require("./resources/blacklistItem");
exports.blacklistItem = blacklistItem_1.blacklistItemType;
var tagAssociation_1 = require("./resources/tagAssociation");
exports.tagAssociation = tagAssociation_1.tagAssociationType;
var product_1 = require("./resources/product");
exports.product = product_1.productType;
