// @lonelyplanet/open-planet-node/dist/scopes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var update = "u";
var read = "r";
var del = "d";
// Dotcom Connect Scopes
exports.openid = "openid";
exports.email = "email";
exports.profile = "profile";
// Open Planet User Scopes
exports.socialAccounts = "social-accounts";
exports.associatedRoles = "associated-roles";
exports.associatedBadges = "associated-badges";
exports.userSelf = "op:users:self";
exports.userSelfRead = exports.userSelf + ":" + read;
exports.userSelfUpdate = exports.userSelf + ":" + update;
exports.userSelfDelete = exports.userSelf + ":" + del;
exports.userSelfSocialAccountsRead = exports.userSelf + ":" + exports.socialAccounts + ":*:" + read;
exports.userSelfSocialAccountsDelete = exports.userSelf + ":" + exports.socialAccounts + ":*:" + del;
exports.userSelfAssociatedRolesRead = exports.userSelf + ":" + exports.associatedRoles + ":*:" + read;
exports.userSelfAssociatedBadgesDelete = exports.userSelf + ":" + exports.associatedBadges + ":*:" + read;
exports.userSelfemailUpdate = exports.userSelf + ":" + exports.email + ":" + update;
exports.userSelfAll = [
    exports.openid,
    exports.email,
    exports.profile,
    exports.userSelfRead,
    exports.userSelfUpdate,
    exports.userSelfDelete,
    exports.userSelfSocialAccountsRead,
    exports.userSelfSocialAccountsDelete,
    exports.userSelfAssociatedRolesRead,
    exports.userSelfAssociatedBadgesDelete,
    exports.userSelfemailUpdate,
];
