// @lonelyplanet/open-planet-node/dist/service.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = y[op[0] & 2 ? "return" : op[0] ? "throw" : "next"]) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [0, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var base_1 = require("./base");
var jsonApiUtils_1 = require("./jsonApiUtils");
var BaseService = /** @class */ (function (_super) {
    __extends(BaseService, _super);
    function BaseService(accessToken) {
        return _super.call(this, accessToken) || this;
    }
    BaseService.prototype.find = function (options) {
        if (options === void 0) { options = {}; }
        return __awaiter(this, void 0, void 0, function () {
            var response, parsed;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.http.fetch("" + this.url(options) + this.qs(options), __assign({}, options, { headers: this.headers() }))];
                    case 1:
                        response = _a.sent();
                        parsed = this.parse(response);
                        return [2 /*return*/, parsed];
                }
            });
        });
    };
    BaseService.prototype.update = function (attributes) {
        return __awaiter(this, void 0, void 0, function () {
            var resource, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        resource = this.mapper.toResource(attributes);
                        Object.keys(resource.attributes).forEach(function (key) {
                            return (typeof resource.attributes[key] === "undefined") && delete resource.attributes[key];
                        });
                        return [4 /*yield*/, this.http.fetch(this.url() + "/" + resource.id, {
                                method: "PATCH",
                                body: {
                                    data: resource,
                                },
                                headers: this.headers(),
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, this.parse(response)];
                }
            });
        });
    };
    BaseService.prototype.create = function (attributes) {
        return __awaiter(this, void 0, void 0, function () {
            var resource, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        resource = this.mapper.toResource(attributes);
                        Object.keys(resource.attributes).forEach(function (key) {
                            return (typeof resource.attributes[key] === "undefined") && delete resource.attributes[key];
                        });
                        return [4 /*yield*/, this.http.fetch("" + this.url(), {
                                method: "POST",
                                body: {
                                    data: resource,
                                },
                                headers: this.headers(),
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, this.parse(response)];
                }
            });
        });
    };
    BaseService.prototype.findById = function (id, options) {
        if (options === void 0) { options = {}; }
        return __awaiter(this, void 0, void 0, function () {
            var baseUrl, url, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        baseUrl = typeof id === "string" ?
                            this.url(options) + "/" + id :
                            this.url(options);
                        url = jsonApiUtils_1.createFindByIdUrl(baseUrl, id, __assign({}, options, { resource: this.resource }));
                        return [4 /*yield*/, this.http.fetch(url, __assign({}, options, { headers: this.headers() }))];
                    case 1:
                        response = _a.sent();
                        if (Array.isArray(response.data)) {
                            return [2 /*return*/, this.parse(response)];
                        }
                        return [2 /*return*/, this.parse(response)];
                }
            });
        });
    };
    BaseService.prototype.destroy = function (id, options) {
        if (options === void 0) { options = {}; }
        return __awaiter(this, void 0, void 0, function () {
            var url, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        url = this.url(options) + "/" + id;
                        return [4 /*yield*/, this.http.fetch(url, {
                                method: "DELETE",
                                headers: this.headers(),
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    BaseService = __decorate([
        inversify_1.injectable(),
        __metadata("design:paramtypes", [Object])
    ], BaseService);
    return BaseService;
}(base_1.default));
exports.default = BaseService;
