// @lonelyplanet/open-planet-node/dist/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Auth = "Auth";
exports.AuthParams = "AuthParams";
exports.Availability = "Availability";
exports.AvailabilityService = "AvailabilityService";
exports.BlogPostMapper = "BlogPostMapper";
exports.BlogPostService = "BlogPostService";
exports.BlogCategoryMapper = "BlogCategoryMapper";
exports.BlogCategoryService = "BlogCategoryService";
exports.Bookmark = "Bookmark";
exports.BookmarkList = "BookmarkList";
exports.BookmarkListService = "BookmarkListService";
exports.BookmarkPopularity = "BookmarkPopularity";
exports.BookmarkPopularityService = "BookmarkPopularityService";
exports.BookmarkService = "BookmarkService";
exports.Config = "Config";
exports.HttpClient = "HttpClient";
exports.IClientFactory = "IClientFactory<IOpenPlanetNode>";
exports.LodgingService = "LodgingService";
exports.OpenPlanetHost = "OpenPlanetHost";
exports.OpenPlanetNode = "OpenPlanetNode";
exports.PartnerActivityService = "PartnerActivityService";
exports.PlaceService = "PlaceService";
exports.PoiService = "PoiService";
exports.TagService = "TagService";
exports.User = "User";
exports.UserResourceMapper = "UserResourceMapper";
exports.UserService = "UserService";
exports.SearchService = "SearchService";
exports.Association = "Association";
exports.AssociationService = "AssociationService";
exports.AssociationEntry = "AssociationEntry";
exports.AssociationEntryService = "AssociationEntryService";
exports.Video = "Video";
exports.VideoService = "VideoService";
exports.VideoSource = "VideoSource";
exports.VideoSourceService = "VideoSourceService";
exports.VideoPlaylist = "VideoPlaylist";
exports.VideoPlaylistService = "VideoPlaylistService";
exports.VideoPlaylistItem = "VideoPlaylistItem";
exports.BlacklistItem = "BlacklistItem";
exports.BlacklistItemService = "BlacklistItemService";
exports.TagAssociation = "TagAssociation";
exports.TagAssociationService = "TagAssociationService";
exports.ProductService = "ProductService";
