// @lonelyplanet/open-planet-node/dist/resources/videoPlaylistItem/mapper.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var _1 = require("./");
var mapper_1 = require("../../mapper");
var mapper_2 = require("../video/mapper");
var video_1 = require("../video");
var VideoPlaylistItemMapper = /** @class */ (function (_super) {
    __extends(VideoPlaylistItemMapper, _super);
    function VideoPlaylistItemMapper() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.videoMapper = new mapper_2.default();
        return _this;
    }
    VideoPlaylistItemMapper.prototype.toModel = function (resource, included) {
        if (included === void 0) { included = []; }
        var model = new _1.default();
        var relationships = resource.relationships, attrs = resource.attributes;
        model.id = resource.id;
        model.createdAt = attrs.create_date;
        model.order = attrs.order;
        if (relationships && relationships.video && relationships.video.data) {
            var data_1 = relationships.video.data;
            var includedResource = included.find(function (i) { return i.type === video_1.videoType && i.id === data_1.id; });
            if (includedResource) {
                model.video = this.videoMapper.toModel(includedResource, included);
            }
            else {
                model.video = data_1;
            }
        }
        return model;
    };
    VideoPlaylistItemMapper.prototype.toResource = function (model) {
        var resource = {
            id: model.id,
            type: _1.videoPlaylistItemType,
            attributes: {
                order: model.order
            },
            relationships: {
                video: {
                    data: null,
                }
            },
        };
        if (model.video) {
            resource.relationships.video.data = {
                id: model.video.id,
                type: video_1.videoType,
            };
        }
        Object.keys(resource.attributes).forEach(function (key) {
            if (typeof resource.attributes[key] === "undefined") {
                delete resource.attributes[key];
            }
        });
        if (Object.keys(resource.relationships).length === 0) {
            delete resource.relationships;
        }
        return resource;
    };
    VideoPlaylistItemMapper = __decorate([
        inversify_1.injectable()
    ], VideoPlaylistItemMapper);
    return VideoPlaylistItemMapper;
}(mapper_1.default));
exports.VideoPlaylistItemMapper = VideoPlaylistItemMapper;
exports.default = mapper_2.default;
