// @lonelyplanet/open-planet-node/src/resources/videoPlaylist/index.ts
import { injectable } from "inversify";
import {
  IJsonApiResponse,
  IJsonApiResource,
  IJsonApiRelationship,
  IJsonApiRelationships,
} from "../../interfaces";
import {IVideoPlaylistItem} from "../videoPlaylistItem/index";

export const videoPlaylistType = "playlist";

export interface IVideoPlaylistAttributes {
  modify_date?: string;
  create_date?: string;
  name?: string;
  slug?: string;
  playlist_type?: string;
  reference_id?: string;
  provider_id?: string;
  provider_name?: string;
}

export interface IVideoPlaylistRelationships extends IJsonApiRelationships {
  items?: IJsonApiRelationship;
}

export interface IVideoPlaylistResource extends IJsonApiResource<IVideoPlaylistAttributes, IVideoPlaylistRelationships> {}
export interface IVideoPlaylistResponse extends IJsonApiResponse<IVideoPlaylistResource> {}

export interface IVideoPlaylist {
  id?: string;
  createdAt?: string;
  updatedAt?: string;
  name?: string;
  slug?: string;
  playlistType?: string;
  referenceId?: string;
  providerId?: string;
  providerName?: string;
  items?: IVideoPlaylistItem[];
}

@injectable()
export default class VideoPlaylist implements IVideoPlaylist {
  id?: string;
  createdAt?: string;
  updatedAt?: string;
  name?: string;
  slug?: string;
  playlistType?: string;
  referenceId?: string;
  providerId?: string;
  providerName?: string;
  items?: IVideoPlaylistItem[];

  constructor(options?: IVideoPlaylist) {
    Object.assign(this, options);
  }
}
