// @lonelyplanet/open-planet-node/dist/resources/tag/topicMapper.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var index_1 = require("./index");
var mapper_1 = require("../../mapper");
var TopicMapper = /** @class */ (function (_super) {
    __extends(TopicMapper, _super);
    function TopicMapper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    TopicMapper.prototype.toModel = function (resource, included) {
        var model = new index_1.Topic();
        model.id = resource.id;
        model.label = resource.attributes.label;
        return model;
    };
    TopicMapper.prototype.toResource = function (model) {
        return {
            id: model.id,
            type: "topic",
            attributes: {
                label: model.label,
            },
            relationships: {}
        };
    };
    return TopicMapper;
}(mapper_1.default));
exports.default = TopicMapper;
