// @lonelyplanet/open-planet-node/src/resources/videoPlaylistItem/index.ts
import { injectable } from "inversify";
import {
  IJsonApiResponse,
  IJsonApiResource,
  IJsonApiRelationship,
  IJsonApiRelationships,
} from "../../interfaces";
import { IVideo } from "../video/index";

export const videoPlaylistItemType = "playlist-item";

export interface IVideoPlaylistItemAttributes {
  order?: number;
  create_date?: string;
}

export interface IVideoPlaylistItemRelationships extends IJsonApiRelationships {
  video?: IJsonApiRelationship;
}

export interface IVideoPlaylistItemResource
  extends IJsonApiResource<IVideoPlaylistItemAttributes, IVideoPlaylistItemRelationships> {}
export interface IVideoPlaylistResponse extends IJsonApiResponse<IVideoPlaylistItemResource> {}

export interface IVideoPlaylistItem {
  id?: string;
  createdAt?: string;
  order?: number;
  video?: IVideo;
}

@injectable()
export default class VideoPlaylistItem implements IVideoPlaylistItem {
  id?: string;
  createdAt?: string;
  order?: number;
  video?: IVideo;

  constructor(options?: IVideoPlaylistItem) {
    Object.assign(this, options);
  }
}
