// backpack-ui/dist/components/card/cardBullets.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _defineProperty2 = require("babel-runtime/helpers/defineProperty");

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _mq = require("../../styles/mq");

var _mq2 = _interopRequireDefault(_mq);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

var _bulletDescription = require("../bulletDescription");

var _bulletDescription2 = _interopRequireDefault(_bulletDescription);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = (0, _defineProperty3.default)({
  marginBottom: "9px"

}, "@media (max-width: " + _mq2.default.max[768] + ")", {
  fontSize: "9px",
  marginBottom: "6px"
});

var CardBullets = function CardBullets(_ref) {
  var bullets = _ref.bullets,
      style = _ref.style;
  return _react2.default.createElement(_bulletDescription2.default, {
    description: bullets,
    style: [styles, style]
  });
};

CardBullets.propTypes = {
  bullets: _propTypes2.default.arrayOf(_propTypes2.default.string).isRequired,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(CardBullets);
