// backpack-ui/dist/components/card/cardHeading.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require("babel-runtime/helpers/extends");

var _extends3 = _interopRequireDefault(_extends2);

var _defineProperty2 = require("babel-runtime/helpers/defineProperty");

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _heading = require("../heading");

var _heading2 = _interopRequireDefault(_heading);

var _typography = require("../../styles/typography");

var _mq = require("../../styles/mq");

var _mq2 = _interopRequireDefault(_mq);

var _timing = require("../../styles/timing");

var _timing2 = _interopRequireDefault(_timing);

var _colors = require("../../styles/colors");

var _colors2 = _interopRequireDefault(_colors);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var mediaQuery = "@media (max-width: " + _mq2.default.max["768"] + ")";

var styles = {
  default: (0, _defineProperty3.default)({
    display: "-webkit-box",
    fontSize: _typography.fontSizeHeading5,
    lineHeight: _typography.lineHeightHeading5,
    maxHeight: "60px",
    overflow: "hidden",
    textOverflow: "ellipsis",
    transition: "color " + _timing2.default.fast + " ease-in-out",
    WebkitBoxOrient: "vertical",
    WebkitLineClamp: 2

  }, mediaQuery, {
    fontSize: _typography.fontSizeHeading7 + "px",
    lineHeight: _typography.lineHeightHeading7
  }),
  dark: {
    color: _colors2.default.textOverlay
  },
  compact: {
    fontSize: _typography.fontSizeHeading7 + "px",
    lineHeight: _typography.lineHeightHeading7
  }
};

var CardHeading = function CardHeading(_ref) {
  var children = _ref.children,
      theme = _ref.theme,
      spacing = _ref.spacing,
      style = _ref.style;
  return _react2.default.createElement(
    _heading2.default,
    {
      level: 3,
      weight: spacing === "compact" ? "normal" : "thin",
      override: (0, _extends3.default)({}, styles.default, theme === "dark" ? styles.dark : {}, spacing === "compact" ? styles.compact : {}, style)
    },
    children
  );
};

CardHeading.propTypes = {
  children: _propTypes2.default.string.isRequired,
  style: _propTypes4.default.style,
  theme: _propTypes2.default.oneOf(["light", "dark"]),
  spacing: _propTypes2.default.oneOf(["normal", "compact"])
};

CardHeading.defaultProps = {
  theme: "light",
  spacing: "normal"
};

exports.default = (0, _radium2.default)(CardHeading);
