// backpack-ui/dist/components/watchLater/watchLaterList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _colors = require("../../styles/colors");

var _colors2 = _interopRequireDefault(_colors);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

var _button = require("../button");

var _button2 = _interopRequireDefault(_button);

var _tileGrid = require("../tileGrid");

var _tileGrid2 = _interopRequireDefault(_tileGrid);

var _tileVideo = require("../tileVideo");

var _tileVideo2 = _interopRequireDefault(_tileVideo);

var _cardVideo = require("../cardVideo");

var _cardVideo2 = _interopRequireDefault(_cardVideo);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  buttonWrapper: {
    width: "100%",
    textAlign: "center"
  },
  button: {
    color: _colors2.default.accentRed
  }
};

var WatchLaterList = function WatchLaterList(_ref) {
  var videos = _ref.videos,
      removeVideo = _ref.removeVideo,
      removeAll = _ref.removeAll,
      style = _ref.style;
  return _react2.default.createElement(
    "div",
    { className: "WatchLaterList", style: style },
    _react2.default.createElement(
      _tileGrid2.default,
      null,
      videos.map(function (video) {
        return _react2.default.createElement(_tileVideo2.default, {
          className: "Tile",
          heading: video.heading,
          bullets: video.bullets,
          runtime: video.runtime,
          actionIcon: "Close",
          onClick: function onClick() {
            return removeVideo(video.id);
          },
          style: {
            marginBottom: "64px"
          },
          imageSrc: video.imageSrc,
          layout: "tile",
          href: video.href
        });
      })
    ),
    _react2.default.createElement(
      "div",
      { style: styles.buttonWrapper },
      _react2.default.createElement(
        _button2.default,
        {
          color: "transparent",
          border: false,
          customStyles: styles.button,
          onClick: removeAll
        },
        "Clear All"
      )
    )
  );
};

WatchLaterList.propTypes = {
  videos: _propTypes2.default.arrayOf(_propTypes2.default.shape(_cardVideo2.default.propTypes)),
  removeVideo: _propTypes2.default.func,
  removeAll: _propTypes2.default.func,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(WatchLaterList);
