// backpack-ui/dist/components/watchLater/watchLaterEmptyList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _defineProperty2 = require("babel-runtime/helpers/defineProperty");

var _defineProperty3 = _interopRequireDefault(_defineProperty2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _colors = require("../../styles/colors");

var _colors2 = _interopRequireDefault(_colors);

var _mq = require("../../styles/mq");

var _mq2 = _interopRequireDefault(_mq);

var _propTypes = require("../../utils/propTypes");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _heading = require("../heading");

var _heading2 = _interopRequireDefault(_heading);

var _button = require("../button");

var _button2 = _interopRequireDefault(_button);

var _link = require("../link");

var _link2 = _interopRequireDefault(_link);

var _icon = require("../icon");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  container: (0, _defineProperty3.default)({
    display: "flex",
    flexDirection: "column",
    justifyContent: "center",
    alignItems: "center",
    position: "absolute",
    left: 0,
    right: 0,
    top: "50%",
    transform: "translateY(-50%)"
  }, "@media (min-width: " + _mq2.default.min[768] + ")", {
    position: "relative",
    transform: "translateY(0)"
  }),
  icon: {
    color: _colors2.default.accentGray,
    marginBottom: "16px",
    fontSize: "34px"
  },
  content: {
    textAlign: "center"
  },
  headline: {
    marginBottom: "16px",
    fontSize: "18px"
  },
  text: {
    fontSize: "14px",
    color: _colors2.default.textSecondary,
    marginBottom: "72px"
  }
};

var WatchLaterEmptyList = function WatchLaterEmptyList(_ref) {
  var style = _ref.style;
  return _react2.default.createElement(
    "div",
    { style: [styles.container, style], className: "WatchLaterEmptyList" },
    _react2.default.createElement(_icon.ClockOutline, { style: styles.icon }),
    _react2.default.createElement(
      "div",
      { style: styles.content },
      _react2.default.createElement(
        _heading2.default,
        {
          level: 3,
          size: "medium",
          weight: "thick",
          override: styles.headline
        },
        "Add some videos"
      ),
      _react2.default.createElement(
        "p",
        { style: styles.text },
        "Find interesting videos and save them for later."
      ),
      _react2.default.createElement(
        _link2.default,
        { to: "/video/v" },
        _react2.default.createElement(
          _button2.default,
          { rounded: true },
          "Browse Channels"
        )
      )
    )
  );
};

WatchLaterEmptyList.propTypes = {
  style: _propTypes2.default.style
};

exports.default = (0, _radium2.default)(WatchLaterEmptyList);
