// backpack-ui/dist/components/watchLater/watchLaterModal.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _modal = require("../modal");

var _modal2 = _interopRequireDefault(_modal);

var _icon = require("../icon");

var _cardVideo = require("../cardVideo");

var _cardVideo2 = _interopRequireDefault(_cardVideo);

var _modalContentWatchLaterList = require("../modalContent/modalContentWatchLaterList");

var _modalContentWatchLaterList2 = _interopRequireDefault(_modalContentWatchLaterList);

var _modalContentSocialAuth = require("../modalContent/modalContentSocialAuth");

var _modalContentSocialAuth2 = _interopRequireDefault(_modalContentSocialAuth);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

var _grid = require("../../utils/grid");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var WatchLaterModal = function WatchLaterModal(_ref) {
  var loggedIn = _ref.loggedIn,
      isOpen = _ref.isOpen,
      onClose = _ref.onClose,
      authMessage = _ref.authMessage,
      videos = _ref.videos,
      removeVideo = _ref.removeVideo,
      removeAll = _ref.removeAll,
      style = _ref.style;
  return _react2.default.createElement(
    _modal2.default,
    {
      isOpen: isOpen,
      leftAction: onClose,
      leftActionContent: _react2.default.createElement(_icon.Close, { width: 24, height: 24 }),
      closeModal: onClose,
      desktopWidth: loggedIn && videos.length > 0 ? "85%" : (0, _grid.span)(6, "static"),
      title: loggedIn ? "Watch Later" : "",
      style: style
    },
    loggedIn ? _react2.default.createElement(_modalContentWatchLaterList2.default, {
      videos: videos,
      removeVideo: removeVideo,
      removeAll: removeAll
    }) : _react2.default.createElement(_modalContentSocialAuth2.default, { message: authMessage })
  );
};

WatchLaterModal.propTypes = {
  loggedIn: _propTypes2.default.bool,
  isOpen: _propTypes2.default.bool,
  onClose: _propTypes2.default.func,
  videos: _propTypes2.default.arrayOf(_propTypes2.default.shape(_cardVideo2.default.propTypes)),
  removeVideo: _propTypes2.default.func,
  removeAll: _propTypes2.default.func,
  authMessage: _propTypes2.default.string,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(WatchLaterModal);
