// @lonelyplanet/open-planet-node/dist/resources/tagAssociation/mapper.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var _1 = require("./");
var mapper_1 = require("../partnerActivity/mapper");
var mapper_2 = require("../poi/mapper");
var mapper_3 = require("../../mapper");
var mappers = {
    partner_activity: new mapper_1.default(),
    poi: new mapper_2.default(),
};
var TagAssociationMapper = /** @class */ (function (_super) {
    __extends(TagAssociationMapper, _super);
    function TagAssociationMapper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    TagAssociationMapper.prototype.toModel = function (resource, included) {
        if (included === void 0) { included = []; }
        var model = new _1.default();
        var attrs = resource.attributes;
        model.id = resource.id;
        model.confidenceScore = attrs.confidence_score;
        model.label = attrs.label;
        model.score = attrs.score;
        var tagAttributes = this.getTagAttributes(this.getTagRelationship(resource.relationships));
        if (tagAttributes) {
            model.tag_type = tagAttributes.tag_type;
            model.tag_id = tagAttributes.tag_id;
        }
        if (resource.relationships.target) {
            var targetResource = resource.relationships.target.data;
            model.target = this.mapIncluded(targetResource, included);
        }
        return model;
    };
    TagAssociationMapper.prototype.toResource = function (model) {
        var resource = {
            id: model.id,
            type: _1.tagAssociationType,
            attributes: {
                confidence_score: model.confidenceScore,
                label: model.label,
                score: model.score,
            },
        };
        resource.relationships = {};
        if (model.target) {
            resource.relationships.target = {
                data: {
                    type: model.target.type,
                    id: model.target.id
                }
            };
        }
        if (model.tag_type) {
            resource.relationships[model.tag_type] = {
                data: {
                    type: model.tag_type,
                    id: model.tag_id,
                }
            };
        }
        Object.keys(resource.attributes).forEach(function (key) {
            if (typeof resource.attributes[key] === "undefined") {
                delete resource.attributes[key];
            }
        });
        if (Object.keys(resource.relationships).length === 0) {
            delete resource.relationships;
        }
        return resource;
    };
    TagAssociationMapper.prototype.getTagRelationship = function (relationships) {
        var tagRelationships = ["topic", "category", "entity", "tag"];
        var tagRelationship = null;
        for (var _i = 0, tagRelationships_1 = tagRelationships; _i < tagRelationships_1.length; _i++) {
            var relationshipName = tagRelationships_1[_i];
            var relationship = relationships[relationshipName];
            if (relationship && typeof relationship.data === "object") {
                tagRelationship = relationships[relationshipName].data;
                break;
            }
        }
        return tagRelationship;
    };
    TagAssociationMapper.prototype.getTagAttributes = function (tagRelationship) {
        if (tagRelationship) {
            return {
                tag_type: tagRelationship.type,
                tag_id: tagRelationship.id,
            };
        }
    };
    TagAssociationMapper.prototype.mapIncluded = function (resourceIdentifier, included) {
        var resource = included.find(function (i) {
            return i.type === resourceIdentifier.type && i.id === resourceIdentifier.id;
        });
        if (!resource) {
            return resourceIdentifier;
        }
        var mapper = mappers[resource.type];
        return mapper.toModel(resource, included);
    };
    TagAssociationMapper = __decorate([
        inversify_1.injectable()
    ], TagAssociationMapper);
    return TagAssociationMapper;
}(mapper_3.default));
exports.TagAssociationMapper = TagAssociationMapper;
exports.default = TagAssociationMapper;
