// @lonelyplanet/open-planet-node/src/resources/blogPost/index.ts
import { injectable } from "inversify";
import {
  IJsonApiResponse,
  IJsonApiResource,
  IJsonApiRelationship,
} from "../../interfaces";

export const blogPostType = "blog-post";
export const blogPostCategoryType = "category";

export interface IBlogPostAttributes {
  attribution: { name: string, title: string }[];
  publication_date: string;
  excerpt: string;
  featured_image: {
    [key: string]: string,
    url: string,
  };
  featured_product: {
    [key: string]: string,
    image_url: string,
    content: string,
  };
  content: { text: string }[];
  seo_metadata: {
    [key: string]: string,
    slug: string,
    title: string,
    description: string,
  };
  title: string;
}

export type IBlogPostRelationships = { categories: IJsonApiRelationship };
export interface IBlogPostResource extends IJsonApiResource<IBlogPostAttributes, IBlogPostRelationships> {}
export interface IUserResponse extends IJsonApiResponse<IBlogPostResource> {}
// export interface IUserSResponse extends IJsonApiCollection<IUserResponse> {}

export interface IBlogPost {
  id: string;
  attribution: { name: string, title: string }[];
  publicationDate: Date;
  excerpt: string;
  featuredImage: {
    [key: string]: string,
    url: string,
  };
  featuredProduct: {
    [key: string]: string,
    image_url: string,
    content: string,
  };
  image: string;
  content: string;
  metadata: {
    [key: string]: string,
    slug: string,
    title: string,
    description: string,
  };
  title: string;
  categories: {
    id: string,
    name: string,
    slug: string
  }[];
}

@injectable()
export default class BlogPost implements IBlogPost {
  static relationships = {
    categories: "categories",
  };

  id: string;
  attribution: { name: string, title: string }[];
  publicationDate: Date;
  image: string;
  excerpt: string;
  featuredImage: {
    [key: string]: string,
    url: string,
  };
  featuredProduct: {
    [key: string]: string,
    image_url: string,
    content: string,
  };
  content: string;
  metadata: {
    [key: string]: string,
    slug: string,
    title: string,
    description: string,
  };
  title: string;
  categories: {
    id: string,
    name: string,
    slug: string
  }[];
}
