// @lonelyplanet/open-planet-node/dist/resources/availability/mapper.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var mapper_1 = require("../../mapper");
var TYPES = require("../../types");
var _1 = require("./");
var AvailabilityMapper = /** @class */ (function (_super) {
    __extends(AvailabilityMapper, _super);
    function AvailabilityMapper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AvailabilityMapper.prototype.toModel = function (resource, included) {
        if (included === void 0) { included = []; }
        var providerId;
        var model = new _1.default();
        var attrs = resource.attributes;
        model.id = resource.id;
        model.available = attrs.available;
        model.offerings = attrs.offerings.map(function (o) { return ({
            maxOccupancy: o.max_occupancy,
            cost: o.cost,
            description: o.description,
        }); }).slice();
        model.provider = attrs.provider;
        model.type = resource.type;
        return model;
    };
    AvailabilityMapper.prototype.toResource = function (model) {
        var resource = {
            id: model.id,
            type: _1.availabilityType,
            attributes: {
                id: model.id,
                available: model.available,
                offerings: model.offerings.map(function (o) { return ({
                    max_occupancy: o.maxOccupancy,
                    cost: o.cost,
                    description: o.description,
                }); }).slice(),
                provider: model.provider,
            },
        };
        return resource;
    };
    __decorate([
        inversify_1.inject(TYPES.Availability),
        __metadata("design:type", Object)
    ], AvailabilityMapper.prototype, "model", void 0);
    AvailabilityMapper = __decorate([
        inversify_1.injectable()
    ], AvailabilityMapper);
    return AvailabilityMapper;
}(mapper_1.default));
exports.default = AvailabilityMapper;
