// backpack-ui/dist/components/bookmarkButtonAlt/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _classnames = require("classnames");

var _classnames2 = _interopRequireDefault(_classnames);

var _listButton = require("../listButton");

var _listButton2 = _interopRequireDefault(_listButton);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BookmarkButtonAlt = function BookmarkButtonAlt(_ref) {
  var onClick = _ref.onClick,
      marked = _ref.marked,
      id = _ref.id,
      className = _ref.className,
      style = _ref.style,
      qaHook = _ref.qaHook;
  return _react2.default.createElement(_listButton2.default, {
    id: id,
    className: (0, _classnames2.default)("BookmarkButtonAlt", className),
    onClick: onClick,
    label: "Save to list",
    icon: marked ? "BookmarkAltActive" : "BookmarkAlt",
    style: style,
    qaHook: qaHook
  });
};

BookmarkButtonAlt.propTypes = {
  onClick: _propTypes2.default.func.isRequired,
  marked: _propTypes2.default.bool,
  id: _propTypes2.default.string,
  className: _propTypes2.default.string,
  style: _propTypes4.default.style,
  qaHook: _propTypes2.default.bool
};

BookmarkButtonAlt.defaultProps = {
  onClick: null,
  marked: false,
  id: null,
  className: null,
  style: null,
  qaHook: false
};

exports.default = (0, _radium2.default)(BookmarkButtonAlt);
