// backpack-ui/dist/hoc/clickOutside/example.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require("babel-runtime/core-js/object/get-prototype-of");

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require("babel-runtime/helpers/classCallCheck");

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require("babel-runtime/helpers/createClass");

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require("babel-runtime/helpers/possibleConstructorReturn");

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require("babel-runtime/helpers/inherits");

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _ = require("./");

var _2 = _interopRequireDefault(_);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ExampleHoc = function (_Component) {
  (0, _inherits3.default)(ExampleHoc, _Component);

  function ExampleHoc(props) {
    (0, _classCallCheck3.default)(this, ExampleHoc);

    var _this = (0, _possibleConstructorReturn3.default)(this, (ExampleHoc.__proto__ || (0, _getPrototypeOf2.default)(ExampleHoc)).call(this, props));

    _this.state = {
      isOpened: false
    };

    _this.onClick = _this.onClick.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(ExampleHoc, [{
    key: "onClick",
    value: function onClick() {
      this.setState({
        isOpened: true
      });
    }
  }, {
    key: "handleClickOutside",
    value: function handleClickOutside() {
      this.setState({
        isOpened: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      return _react2.default.createElement(
        "div",
        { ref: this.props.innerRef },
        _react2.default.createElement(
          "button",
          { onClick: this.onClick },
          "Toggle"
        ),
        this.state.isOpened && _react2.default.createElement(
          "div",
          null,
          "Menu content"
        )
      );
    }
  }]);
  return ExampleHoc;
}(_react.Component);

ExampleHoc.propTypes = {
  innerRef: _propTypes2.default.func.isRequired
};

exports.default = (0, _2.default)(ExampleHoc);
