// @lonelyplanet/travel-agent/dist/middleware/prometheus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var promBundle = require("express-prom-bundle");
var pathToRegexp = require("path-to-regexp");
;
var prometheusHttpRequestBuckets = [
    0.01,
    0.05,
    0.1,
    0.2,
    0.3,
    0.4,
    0.5,
    0.6,
    0.7,
    0.8,
    0.9,
    1.0,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    10,
];
function customNormalize(url, routes, options) {
    var path = url.split("?")[0];
    // count all docs (no matter which file) as a single url
    for (var _i = 0, routes_1 = routes; _i < routes_1.length; _i++) {
        var route = routes_1[_i];
        var routeRegex = pathToRegexp(route.route);
        if (routeRegex.exec(path)) {
            return route.name;
        }
    }
    return options && options.defaultPath ? options.defaultPath : url;
}
exports.customNormalize = customNormalize;
function createPrometheusMiddleware(_a) {
    var _b = _a.routes, routes = _b === void 0 ? [] : _b, defaultPath = _a.defaultPath;
    promBundle.normalizePath = function (req) {
        var url = req.url;
        return customNormalize(url, routes, {
            defaultPath: defaultPath,
        });
    };
    promBundle.promClient.register.setDefaultLabels({ direction: "inbound" });
    promBundle.promClient.collectDefaultMetrics();
    var prometheus = promBundle({
        includePath: true,
        includeMethod: true,
        excludeRoutes: [/assets/],
        buckets: prometheusHttpRequestBuckets,
    });
    return prometheus;
}
exports.default = createPrometheusMiddleware;
;
