// @lonelyplanet/open-planet-node/dist/resources/bookmarkPopularity/mapper.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var mapper_1 = require("../../mapper");
var TYPES = require("../../types");
var bookmarkPopularity_1 = require("../bookmarkPopularity");
var BookmarkPopularityMapper = /** @class */ (function (_super) {
    __extends(BookmarkPopularityMapper, _super);
    function BookmarkPopularityMapper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BookmarkPopularityMapper.prototype.toModel = function (resource) {
        var model = new bookmarkPopularity_1.default();
        model.id = resource.id;
        model.count = resource.attributes.bookmark_count;
        model.target = resource.relationships.target.data;
        return model;
    };
    BookmarkPopularityMapper.prototype.toResource = function (model) {
        var resource = {
            id: model.id,
            type: bookmarkPopularity_1.bookmarkPopularityType,
            attributes: {
                bookmark_count: model.count,
            },
            relationships: {}
        };
        if (model.target) {
            resource.relationships.target = {
                data: {
                    type: model.target.type,
                    id: model.target.id,
                }
            };
        }
        return resource;
    };
    __decorate([
        inversify_1.inject(TYPES.BookmarkPopularity),
        __metadata("design:type", Object)
    ], BookmarkPopularityMapper.prototype, "model", void 0);
    BookmarkPopularityMapper = __decorate([
        inversify_1.injectable()
    ], BookmarkPopularityMapper);
    return BookmarkPopularityMapper;
}(mapper_1.default));
exports.BookmarkPopularityMapper = BookmarkPopularityMapper;
exports.default = BookmarkPopularityMapper;
