// backpack-ui/dist/components/form/heightExpander.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _getPrototypeOf = require("babel-runtime/core-js/object/get-prototype-of");

var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);

var _classCallCheck2 = require("babel-runtime/helpers/classCallCheck");

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require("babel-runtime/helpers/createClass");

var _createClass3 = _interopRequireDefault(_createClass2);

var _possibleConstructorReturn2 = require("babel-runtime/helpers/possibleConstructorReturn");

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require("babel-runtime/helpers/inherits");

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var HeightExpander = function (_React$Component) {
  (0, _inherits3.default)(HeightExpander, _React$Component);

  function HeightExpander(props) {
    (0, _classCallCheck3.default)(this, HeightExpander);

    var _this = (0, _possibleConstructorReturn3.default)(this, (HeightExpander.__proto__ || (0, _getPrototypeOf2.default)(HeightExpander)).call(this, props));

    _this.state = {
      height: props.baseHeight
    };

    _this.updateHeight = _this.updateHeight.bind(_this);
    return _this;
  }

  (0, _createClass3.default)(HeightExpander, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.element = document.getElementById(this.props.idToFind);
      window.addEventListener("resize", this.updateHeight);
      this.updateHeight();
    }
  }, {
    key: "componentWillReceiveProps",
    value: function componentWillReceiveProps() {
      this.updateHeight();
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      window.removeEventListener("resize", this.updateHeight);
    }
  }, {
    key: "updateHeight",
    value: function updateHeight() {
      var _this2 = this;

      this.setState({
        height: "1px"
      }, function () {
        _this2.setState({
          height: _this2.element.scrollHeight + "px"
        });
      });
    }
  }, {
    key: "render",
    value: function render() {
      return this.props.children(this.updateHeight, this.state.height);
    }
  }]);
  return HeightExpander;
}(_react2.default.Component);

HeightExpander.propTypes = {
  children: _propTypes2.default.func.isRequired,
  baseHeight: _propTypes2.default.string.isRequired,
  idToFind: _propTypes2.default.string.isRequired
};

HeightExpander.defaultProps = {
  baseHeight: "auto"
};

exports.default = HeightExpander;
