// backpack-ui/dist/components/cardShelf/cardShelfHeader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _link = require("../link");

var _link2 = _interopRequireDefault(_link);

var _heading = require("../heading");

var _heading2 = _interopRequireDefault(_heading);

var _moreLink = require("../moreLink");

var _moreLink2 = _interopRequireDefault(_moreLink);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  container: {
    marginBottom: "19px",
    position: "relative"
  },

  heading: {
    display: "inline-block",
    fontSize: "26px"
  },

  moreLink: {
    bottom: "15px",
    fontWeight: 400,
    position: "absolute",
    right: 0
  }
};

var CardShelfHeader = function CardShelfHeader(_ref) {
  var heading = _ref.heading,
      href = _ref.href,
      adSlot = _ref.adSlot,
      style = _ref.style;
  return _react2.default.createElement(
    "header",
    {
      className: "CardShelf-header",
      style: [styles.container, style]
    },
    _react2.default.createElement(
      _heading2.default,
      {
        level: 2,
        weight: "thin",
        override: styles.heading
      },
      heading
    ),
    adSlot,
    href && _react2.default.createElement(
      _link2.default,
      { to: href },
      _react2.default.createElement(
        _moreLink2.default,
        {
          style: styles.moreLink,
          size: "small",
          caps: true
        },
        "View more"
      )
    )
  );
};

CardShelfHeader.propTypes = {
  heading: _propTypes2.default.string.isRequired,
  href: _propTypes2.default.string,
  adSlot: _propTypes2.default.element,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(CardShelfHeader);
