// backpack-ui/dist/components/navigation/navigationTab.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _objectWithoutProperties2 = require("babel-runtime/helpers/objectWithoutProperties");

var _objectWithoutProperties3 = _interopRequireDefault(_objectWithoutProperties2);

var _extends2 = require("babel-runtime/helpers/extends");

var _extends3 = _interopRequireDefault(_extends2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _colors = require("../../styles/colors");

var _colors2 = _interopRequireDefault(_colors);

var _timing = require("../../styles/timing");

var _timing2 = _interopRequireDefault(_timing);

var _typography = require("../../styles/typography");

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

var _mixins = require("../../utils/mixins");

var _typography2 = require("../../utils/typography");

var _createQAHook = require("../../utils/createQAHook");

var _createQAHook2 = _interopRequireDefault(_createQAHook);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = (0, _extends3.default)({}, {
  backgroundColor: _colors2.default.bgPrimary,
  borderBottomColor: _colors2.default.bgPrimary,
  borderBottomStyle: "solid",
  borderBottomWidth: "4px",
  color: _colors2.default.textPrimary,
  cursor: "pointer",
  display: "inline-block",
  fontWeight: _typography.fontWeightMedium,
  height: "100%",
  paddingLeft: "16px",
  paddingRight: "16px",
  paddingTop: "4px", // offset border width, keep text centered
  textAlign: "center",
  transition: "border-bottom-color " + _timing2.default.fast + " ease-in-out,\n    color " + _timing2.default.fast + " ease-in-out",
  verticalAlign: "middle",

  ":hover": {
    color: _colors2.default.textSecondary
  },

  ":active": {
    color: _colors2.default.textSecondary
  },

  ":focus": (0, _mixins.outline)()
}, (0, _typography2.textUppercase)());

var NavigationTab = function NavigationTab(props) {
  var children = props.children,
      qaHook = props.qaHook,
      onClick = props.onClick,
      active = props.active,
      style = props.style,
      label = props.label,
      properties = (0, _objectWithoutProperties3.default)(props, ["children", "qaHook", "onClick", "active", "style", "label"]);


  return _react2.default.createElement(
    "button",
    (0, _extends3.default)({
      className: "NavigationTab",
      onClick: onClick,
      "data-testid": qaHook ? (0, _createQAHook2.default)(label, "NavigationTab", "btn") : null,
      style: [styles, active && { borderBottomColor: _colors2.default.linkPrimary }, style]
    }, properties),
    children
  );
};

NavigationTab.propTypes = {
  children: _propTypes2.default.string.isRequired,
  onClick: _propTypes2.default.func,
  active: _propTypes2.default.bool,
  style: _propTypes4.default.style,
  label: _propTypes2.default.string,
  qaHook: _propTypes2.default.bool
};

NavigationTab.defaultProps = {
  qaHook: false
};

exports.default = (0, _radium2.default)(NavigationTab);
