// @lonelyplanet/open-planet-node/dist/resources/product/index.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
exports.productType = "product";
var Product = /** @class */ (function () {
    function Product(options) {
        this.type = "product";
        this.descriptions = {};
        this.images = {
            original: "",
            small: [],
            medium: [],
            large: [],
        };
        this.pricing = [];
        Object.assign(this, options);
    }
    Product = __decorate([
        inversify_1.injectable(),
        __metadata("design:paramtypes", [Object])
    ], Product);
    return Product;
}());
exports.default = Product;
