// @lonelyplanet/open-planet-node/dist/utils/imgix.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var providers = {
    bdc: {
        baseUrl: "aff.bstatic.com",
        domain: "https://lpbookingcom.imgix.net",
        regex: /http:\/\/aff.bstatic.com/,
    },
    gadventures: {
        baseUrl: "media.gadventures.com",
        domain: "https://lpgadventures.imgix.net",
        regex: /https:\/\/media.gadventures.com/,
    },
    hw: {
        baseUrl: "ucd.hwstatic.com",
        domain: "https://lphostelworld.imgix.net",
        regex: /http:\/\/(.*)hwstatic.com/,
    },
    media: {
        baseUrl: "media.lonelyplanet.com",
        domain: "https://lonelyplanetimages.imgix.net",
        regex: /https:\/\/media.lonelyplanet.com/,
    },
    viator: {
        baseUrl: "cache-graphicslib.viator.com",
        domain: "https://lpviator.imgix.net",
        regex: /http:\/\/cache-graphicslib.viator.com/,
    },
};
function imgix(url, provider) {
    if (provider === void 0) { provider = "media"; }
    if (Object.keys(providers).indexOf(provider) === -1) {
        provider = "media";
    }
    if (!providers[provider].regex.test(url)) {
        return url;
    }
    return url.replace(providers[provider].regex, providers[provider].domain);
}
exports.default = imgix;
