// backpack-ui/dist/components/cardBasic/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _card = require("../card");

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CardBasic = function CardBasic(_ref) {
  var href = _ref.href,
      imageSrc = _ref.imageSrc,
      heading = _ref.heading,
      bullets = _ref.bullets,
      style = _ref.style;
  return _react2.default.createElement(
    _card.Card,
    { style: style },
    _react2.default.createElement(_card.CardImage, {
      href: href,
      src: imageSrc
    }),
    _react2.default.createElement(
      _card.CardText,
      null,
      _react2.default.createElement(
        _card.CardAnchor,
        { href: href },
        bullets && bullets.length > 0 && _react2.default.createElement(_card.CardBullets, { bullets: bullets }),
        _react2.default.createElement(
          _card.CardHeading,
          null,
          heading
        )
      )
    )
  );
};

CardBasic.propTypes = {
  href: _propTypes2.default.string.isRequired,
  imageSrc: _propTypes2.default.string.isRequired,
  heading: _propTypes2.default.string.isRequired,
  bullets: _propTypes2.default.arrayOf(_propTypes2.default.string).isRequired,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(CardBasic);
