// @lonelyplanet/dotcom-core/dist/containers/asyncComponent/index.js
"use strict";

var __extends = this && this.__extends || function () {
  var extendStatics = Object.setPrototypeOf || {
    __proto__: []
  } instanceof Array && function (d, b) {
    d.__proto__ = b;
  } || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
  };

  return function (d, b) {
    extendStatics(d, b);

    function __() {
      this.constructor = d;
    }

    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
  };
}();

var __assign = this && this.__assign || Object.assign || function (t) {
  for (var s, i = 1, n = arguments.length; i < n; i++) {
    s = arguments[i];

    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
  }

  return t;
};

Object.defineProperty(exports, "__esModule", {
  value: true
});

var React = require("react");

function asyncComponent(getComponent) {
  var ComponentCache = null;
  return (
    /** @class */
    function (_super) {
      __extends(AsyncComponent, _super);

      function AsyncComponent(props) {
        var _this = _super.call(this, props) || this;

        _this.state = {
          Component: ComponentCache
        };
        return _this;
      }

      AsyncComponent.prototype.componentWillMount = function () {
        var _this = this;

        if (!this.state.Component) {
          getComponent().then(function (Component) {
            ComponentCache = Component;

            _this.setState({
              Component: Component
            });
          });
        }
      };

      AsyncComponent.prototype.render = function () {
        var Component = this.state.Component;

        if (Component) {
          return React.createElement(Component, __assign({}, this.props));
        }

        return null;
      };

      return AsyncComponent;
    }(React.Component)
  );
}

exports.default = asyncComponent;
