// @lonelyplanet/travel-agent/dist/webpack/config.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/* tslint:disable: no-var-requires */
var path = require("path");
var webpack = require("webpack");
var common_1 = require("./common");
var notifier_1 = require("./notifier");
var userConfig_1 = require("./userConfig");
var webpackMerge = require("webpack-merge");
var loaders_1 = require("./loaders");
var config = {
    entry: {
        common: ["webpack-hot-middleware/client"],
    },
    module: {
        rules: loaders_1.getLoaders("dev").slice(),
    },
    output: {
        chunkFilename: "[name]-chunk.js",
        filename: "[name].js",
        path: path.join(process.cwd(), "public", "assets"),
        publicPath: "/assets/",
    },
    plugins: common_1.commonPlugins.concat([
        new webpack.HotModuleReplacementPlugin(),
        new notifier_1.default(),
        new webpack.WatchIgnorePlugin([/\.d\.ts$/]),
    ]),
    resolve: {
        extensions: [".tsx", ".ts", ".js"],
        modules: ["node_modules", path.join(__dirname, "..", "..", "node_modules")],
    },
    resolveLoader: {
        extensions: [".tsx", ".ts", ".js"],
        modules: ["node_modules", path.join(__dirname, "..", "..", "node_modules")],
    },
};
exports.default = webpackMerge(config, userConfig_1.userConfig);
