// backpack-ui/dist/components/settingBlock/settingBlockHeader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _colors = require("../../styles/colors");

var _colors2 = _interopRequireDefault(_colors);

var _typography = require("../../styles/typography");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  label: {
    display: "block",
    fontSize: _typography.fontSizeHeading7 + "px",
    fontWeight: _typography.fontWeightMedium,
    lineHeight: _typography.lineHeightReset,
    paddingTop: "16px"
  },

  subtitle: {
    color: _colors2.default.textSecondary,
    fontSize: _typography.fontSizeUppercase + "px",
    fontWeight: _typography.fontWeightLight,
    marginTop: "4px"
  }
};

var SettingBlockHeader = function SettingBlockHeader(_ref) {
  var children = _ref.children,
      htmlFor = _ref.htmlFor,
      subtitle = _ref.subtitle;
  return _react2.default.createElement(
    "div",
    { className: "SettingBlockHeader" },
    _react2.default.createElement(
      "label",
      { htmlFor: htmlFor, style: styles.label },
      children
    ),
    subtitle && _react2.default.createElement(
      "p",
      { style: styles.subtitle },
      subtitle
    )
  );
};

SettingBlockHeader.propTypes = {
  children: _propTypes2.default.string.isRequired,
  htmlFor: _propTypes2.default.string,
  subtitle: _propTypes2.default.string
};

exports.default = (0, _radium2.default)(SettingBlockHeader);
