// backpack-ui/dist/components/settingBlock/settingBlockButtonWrapper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require("babel-runtime/helpers/extends");

var _extends3 = _interopRequireDefault(_extends2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _mixins = require("../../utils/mixins");

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

var _validReactAttributes = require("../../utils/validReactAttributes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  container: {
    color: "inherit",
    cursor: "pointer",
    ":focus": {
      outline: "none"
    }
  },

  buttonResets: {
    backgroundColor: "transparent",
    textAlign: "left",
    width: "100%"
  }
};

var scopedStyles = {
  ".SettingBlockButtonWrapper:focus .Checkbox": (0, _mixins.outline)()
};

var SettingBlockButtonWrapper = function SettingBlockButtonWrapper(props) {
  var onClick = props.onClick;

  var sanitizedProps = (0, _validReactAttributes.validReactAttributes)(props);

  return _react2.default.createElement(
    "button",
    (0, _extends3.default)({}, sanitizedProps, {
      onClick: onClick,
      className: "SettingBlockButtonWrapper",
      style: [styles.container, styles.buttonResets, props.style]
    }),
    _react2.default.createElement(_radium.Style, { rules: scopedStyles }),
    props.children
  );
};

SettingBlockButtonWrapper.propTypes = {
  children: _propTypes2.default.element,
  style: _propTypes4.default.style,
  onClick: _propTypes2.default.func
};

exports.default = (0, _radium2.default)(SettingBlockButtonWrapper);
