// backpack-ui/dist/components/settingBlock/settingBlock.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _settingBlockWrapper = require("../settingBlock/settingBlockWrapper");

var _settingBlockWrapper2 = _interopRequireDefault(_settingBlockWrapper);

var _settingBlockHeader = require("../settingBlock/settingBlockHeader");

var _settingBlockHeader2 = _interopRequireDefault(_settingBlockHeader);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SettingBlock = function SettingBlock(_ref) {
  var children = _ref.children,
      error = _ref.error,
      title = _ref.title,
      subtitle = _ref.subtitle,
      htmlFor = _ref.htmlFor,
      style = _ref.style;
  return _react2.default.createElement(
    _settingBlockWrapper2.default,
    { style: style, error: error },
    _react2.default.createElement(
      _settingBlockHeader2.default,
      {
        htmlFor: htmlFor,
        subtitle: subtitle
      },
      title
    ),
    children
  );
};

SettingBlock.propTypes = {
  children: _propTypes2.default.element.isRequired,
  title: _propTypes2.default.string.isRequired,
  error: _propTypes2.default.bool,
  subtitle: _propTypes2.default.string,
  htmlFor: _propTypes2.default.string,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(SettingBlock);
