// backpack-ui/dist/components/settingBlock/settingBlockWrapper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _colors = require("../../styles/colors");

var _colors2 = _interopRequireDefault(_colors);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  hasBorder: {
    borderBottomColor: _colors2.default.borderPrimary,
    borderBottomStyle: "solid",
    borderBottomWidth: "1px"
  },

  hasAction: {
    alignItems: "center",
    display: "flex",
    justifyContent: "space-between"
  },

  error: {
    borderBottomColor: _colors2.default.accentRed
  }
};

var SettingBlockWrapper = function SettingBlockWrapper(_ref) {
  var children = _ref.children,
      error = _ref.error,
      hasAction = _ref.hasAction,
      hasBorder = _ref.hasBorder,
      style = _ref.style;
  return _react2.default.createElement(
    "div",
    {
      className: "SettingBlockWrapper",
      style: [error && styles.error, hasBorder && styles.hasBorder, hasAction && styles.hasAction, style]
    },
    children
  );
};

SettingBlockWrapper.propTypes = {
  children: _propTypes2.default.arrayOf(_propTypes2.default.element).isRequired,
  error: _propTypes2.default.bool,
  hasAction: _propTypes2.default.bool,
  hasBorder: _propTypes2.default.bool,
  style: _propTypes4.default.style
};

SettingBlockWrapper.defaultProps = {
  hasBorder: true
};

exports.default = (0, _radium2.default)(SettingBlockWrapper);
