// backpack-ui/dist/components/settingBlock/settingBlockButtonDescription.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

var _zIndex = require("../../styles/zIndex");

var _zIndex2 = _interopRequireDefault(_zIndex);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  container: {
    flexBasis: "60%",
    paddingBottom: "16px",
    zIndex: _zIndex2.default.below
  }
};

var SettingBlockButtonDescription = function SettingBlockButtonDescription(_ref) {
  var children = _ref.children,
      style = _ref.style;
  return _react2.default.createElement(
    "div",
    {
      className: "SettingBlockButtonDescription",
      style: [styles.container, style]
    },
    children
  );
};

SettingBlockButtonDescription.propTypes = {
  children: _propTypes2.default.arrayOf(_propTypes2.default.oneOfType([_propTypes2.default.element, _propTypes2.default.string])),
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(SettingBlockButtonDescription);
