// backpack-ui/dist/components/settingBlock/settingBlockDescription.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _typography = require("../../styles/typography");

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  container: {
    fontSize: _typography.fontSizeHeading7 + "px",
    fontWeight: _typography.fontWeightLight,
    marginTop: "8px"
  }
};

var SettingBlockDescription = function SettingBlockDescription(_ref) {
  var children = _ref.children,
      style = _ref.style;
  return _react2.default.createElement(
    "p",
    {
      className: "SettingBlockDescription",
      style: [styles.container, style]
    },
    children
  );
};

SettingBlockDescription.propTypes = {
  children: _propTypes2.default.string,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(SettingBlockDescription);
