// backpack-ui/dist/components/settingBlock/settingBlockSection.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _colors = require("../../styles/colors");

var _colors2 = _interopRequireDefault(_colors);

var _typography = require("../../styles/typography");

var _color = require("../../utils/color");

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

var _categoryLabel = require("../categoryLabel");

var _categoryLabel2 = _interopRequireDefault(_categoryLabel);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  container: {
    marginBottom: "40px"
  },

  sectionHeading: {
    color: _colors2.default.textSecondary,
    fontWeight: _typography.fontWeightMedium,
    paddingBottom: "16px",
    verticalAlign: "bottom"
  },

  inputPlaceholderRules: {
    "::-webkit-input-placeholder": {
      color: (0, _color.rgba)(_colors2.default.textPrimary, 0.3)
    },

    "::-moz-placeholder": {
      color: (0, _color.rgba)(_colors2.default.textPrimary, 0.3)
    },

    ":-ms-input-placeholder": {
      color: (0, _color.rgba)(_colors2.default.textPrimary, 0.3)
    }
  }
};

var SettingBlockSection = function SettingBlockSection(_ref) {
  var children = _ref.children,
      heading = _ref.heading,
      style = _ref.style;
  return _react2.default.createElement(
    "div",
    {
      className: "SettingBlockSection",
      style: [styles.container, style]
    },
    _react2.default.createElement(_radium.Style, {
      scopeSelector: ".SettingBlockSection",
      rules: styles.inputPlaceholderRules
    }),
    _react2.default.createElement(
      _categoryLabel2.default,
      { style: styles.sectionHeading, light: true },
      heading
    ),
    children
  );
};

SettingBlockSection.propTypes = {
  children: _propTypes2.default.arrayOf(_propTypes2.default.element).isRequired,
  heading: _propTypes2.default.string.isRequired,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(SettingBlockSection);
