// backpack-ui/dist/components/settingBlock/settingBlockListItemWrapper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  container: {
    marginBottom: "8px"
  }
};

var SettingBlockListItemWrapper = function SettingBlockListItemWrapper(_ref) {
  var children = _ref.children,
      style = _ref.style;
  return _react2.default.createElement(
    "div",
    {
      className: "SettingBlockListItemWrapper",
      style: [styles.container, style]
    },
    children
  );
};

SettingBlockListItemWrapper.propTypes = {
  children: _propTypes2.default.element,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(SettingBlockListItemWrapper);
