// @lonelyplanet/open-planet-node/dist/__tests__/blacklistItemService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var service_1 = require("../resources/blacklistItem/service");
var blacklistItem_1 = require("../resources/blacklistItem");
describe("BlacklistItemService", function () {
    it("should fetch a blacklistItem by given id", function () {
        var fetchMock = jest.fn(function () { return new Promise((function (resolve) { return resolve({
            data: {
                attributes: {
                    "created_at": "2018-05-23T10:46:56.337288",
                    "updated_at": "2018-05-23T10:46:56.337288",
                },
                id: "c41e0fde-5e64-11e8-b625-2fe248649608",
                type: "blacklist-item",
                relationships: {
                    target: {
                        data: {
                            id: "g-SPIT",
                            type: "partner_activity"
                        },
                        links: {
                            related: "/partner-activities/g-SPIT"
                        }
                    }
                }
            }
        }); })); });
        var service = new service_1.default("access123");
        service.http = {
            fetch: fetchMock,
        };
        service.findById("c41e0fde-5e64-11e8-b625-2fe248649608")
            .then(function (blacklistItem) {
            expect(blacklistItem.id).toBe("c41e0fde-5e64-11e8-b625-2fe248649608");
        })
            .catch(function (e) { return console.error(e); });
        expect(fetchMock).toHaveBeenCalledWith("/blacklist-items/c41e0fde-5e64-11e8-b625-2fe248649608", { "headers": { "Authorization": "Bearer access123" } });
    });
    it("should create a blacklist-item with given attributes", function () {
        var fetchMock = jest.fn(function () { return new Promise((function (resolve) { return resolve({
            data: {
                attributes: {
                    "created_at": "2018-05-23T10:46:56.337288",
                    "updated_at": "2018-05-23T10:46:56.337288",
                },
                id: "c41e0fde-5e64-11e8-b625-2fe248649608",
                type: "blacklist-item",
                relationships: {
                    target: {
                        data: {
                            id: "351223",
                            type: "poi"
                        },
                        links: {
                            related: "/pois/351223"
                        }
                    }
                }
            }
        }); })); });
        var service = new service_1.default("access123");
        service.http = {
            fetch: fetchMock,
        };
        var blacklistItem = new blacklistItem_1.default({
            "target": {
                id: "351223",
                type: "poi"
            }
        });
        service.create(blacklistItem).then(function (result) {
            expect(result.id).toBe("c41e0fde-5e64-11e8-b625-2fe248649608");
            expect(result.target.id).toBe("351223");
            expect(result.target.type).toBe("poi");
        })
            .catch(function (e) { return console.error(e); });
        expect(fetchMock).toHaveBeenCalledWith("/blacklist-items", {
            "body": {
                "data": {
                    "attributes": {},
                    "relationships": {
                        "target": {
                            "data": { "id": "351223", "type": "poi" }
                        }
                    },
                    "type": "blacklist-item"
                }
            },
            "headers": { "Authorization": "Bearer access123" },
            "method": "POST"
        });
    });
    it("should delete a blacklist-item with given id", function () {
        var fetchMock = jest.fn(function () { return new Promise((function (resolve) { return resolve({
            data: {}
        }); })); });
        var service = new service_1.default("access123");
        service.http = {
            fetch: fetchMock,
        };
        service.destroy("c41e0fde-5e64-11e8-b625-2fe248649608");
        expect(fetchMock).toHaveBeenCalledWith("/blacklist-items/c41e0fde-5e64-11e8-b625-2fe248649608", {
            "headers": {
                "Authorization": "Bearer access123"
            },
            "method": "DELETE"
        });
    });
});
