// @lonelyplanet/open-planet-node/dist/__tests__/poiMapper.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var mapper_1 = require("../resources/poi/mapper");
describe("PoiMapper", function () {
    it("should map a POI entry to a POI model", function () {
        var mapper = new mapper_1.default();
        var model = mapper.toModel({
            type: "poi",
            id: "1234",
            attributes: {
                name: "Ryman",
                published_at: "some date",
                score: 3,
                alternative_names: ["Mother church"],
                telephone: {
                    callingCode: "123",
                    national: "regional",
                    numbers: ['1', '2', '3'],
                },
            },
            relationships: {
                "containing-place": {
                    data: {
                        type: "place",
                        id: "123456"
                    }
                }
            },
        }, [
            {
                id: "123456",
                attributes: {
                    name: "Downtown",
                    place_type: "neighborhood"
                },
                relationships: {
                    ancestry: {
                        data: [
                            { id: "362228", type: "place" },
                            { id: "298234", type: "place" },
                            { id: "11001010", type: "place" },
                            { id: "11001011", type: "place" },
                            { id: "11001100", type: "place" },
                            { id: "07041776", type: "place" }
                        ],
                        links: {}
                    }
                },
                type: "place"
            },
            {
                id: "362228",
                attributes: {
                    name: "Nashville",
                    place_type: "city"
                },
                type: "place"
            },
            {
                id: "298234",
                attributes: {
                    name: "North America",
                    place_type: "continent"
                },
                type: "place"
            },
            {
                id: "07041776",
                attributes: {
                    name: "United States",
                    place_type: "country",
                    calling_code: "10181",
                },
                type: "place"
            },
            {
                id: "11001010",
                attributes: {
                    name: "Tennessee",
                    place_type: "region"
                },
                type: "place"
            },
            {
                id: "11001011",
                attributes: {
                    name: "Middle Tennessee",
                    place_type: "region"
                },
                type: "place"
            },
            {
                id: "11001100",
                attributes: {
                    name: "Americas",
                    place_type: "region"
                },
                type: "place"
            },
            {
                id: "9999",
                attributes: {
                    path: "/a/g/hi/t/d1f3f73703867dfd0c11f047927181e6-country-music-hall-of-fame-museum.jpg"
                },
                type: "image",
                links: {
                    pixels: "https://media.lonelyplanet.com/a/g/hi/t/d1f3f73703867dfd0c11f047927181e6-country-music-hall-of-fame-museum.jpg",
                    self: "/images/9999"
                }
            },
            {
                id: "8888",
                type: "image_association",
                attributes: {},
                relationships: {
                    from: {
                        data: {
                            id: "9999",
                            type: "image"
                        }
                    },
                    to: {
                        data: {
                            id: "1234",
                            type: "poi"
                        }
                    }
                }
            }
        ]);
        expect(model.name).toEqual("Ryman");
        expect(model.alternative_names).toEqual(["Mother church"]);
        expect(model.publishedAt).toEqual("some date");
        expect(model.lpInternalScore).toEqual(3);
        expect(model.containingPlaceName).toEqual("Downtown");
        expect(model.containingCityName).toEqual("Nashville");
        expect(model.containingContinentName).toEqual("North America");
        expect(model.containingCountryName).toEqual("United States");
        expect(model.telephone.callingCode).toEqual("+10181");
        expect(model.containingNeighborhoodName).toEqual("Downtown");
        expect(model.containingRegionNames).toEqual(["Tennessee", "Middle Tennessee", "Americas"]);
        expect(model.images.length).toBe(1);
        expect(model.images[0].path).toBe("/a/g/hi/t/d1f3f73703867dfd0c11f047927181e6-country-music-hall-of-fame-museum.jpg");
        expect(model.images[0].url).toBe("https://lonelyplanetimages.imgix.net/a/g/hi/t/d1f3f73703867dfd0c11f047927181e6-country-music-hall-of-fame-museum.jpg");
    });
});
