// @lonelyplanet/open-planet-node/dist/resources/search/service.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = y[op[0] & 2 ? "return" : op[0] ? "throw" : "next"]) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [0, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var base_1 = require("../../base");
var _1 = require("./");
var mapper_1 = require("./mapper");
var mapper_2 = require("./autocomplete/mapper");
var SearchService = /** @class */ (function (_super) {
    __extends(SearchService, _super);
    function SearchService() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.resource = _1.searchType;
        _this.mapper = new mapper_1.default();
        return _this;
    }
    SearchService.prototype.url = function () {
        return this.host + "/search";
    };
    SearchService.prototype.query = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            var response, parsed;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.http.fetch("" + this.url() + this.qs(options), __assign({}, options, { headers: this.headers() }))];
                    case 1:
                        response = _a.sent();
                        parsed = this.parse(response);
                        return [2 /*return*/, {
                                data: parsed,
                                meta: {
                                    query: response.meta.query,
                                    total: response.meta.total,
                                },
                                total: response.meta.total,
                            }];
                }
            });
        });
    };
    SearchService.prototype.createClickthrough = function (term, searchResultId) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.http.fetch(this.url() + "/clickthroughs", {
                            method: "POST",
                            body: {
                                data: {
                                    attributes: { term: term, id: searchResultId },
                                    type: 'search-clickthrough',
                                }
                            },
                            headers: this.headers(),
                        })];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    SearchService.prototype.createAutocompleteClickthrough = function (prefix, searchResultSwiftypeId) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.http.fetch(this.url() + "/autocomplete/clickthroughs", {
                            method: "POST",
                            body: {
                                data: {
                                    attributes: { prefix: prefix, id: searchResultSwiftypeId },
                                    type: 'autocomplete-clickthrough',
                                }
                            },
                            headers: this.headers(),
                        })];
                    case 1:
                        _a.sent();
                        return [2 /*return*/];
                }
            });
        });
    };
    SearchService.prototype.queryAutocomplete = function (options) {
        return __awaiter(this, void 0, void 0, function () {
            var url, response, mapper;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        url = this.url() + "/autocomplete" + this.qs(options);
                        return [4 /*yield*/, this.http.fetch(url, __assign({ headers: this.headers() }, options))];
                    case 1:
                        response = _a.sent();
                        mapper = new mapper_2.default();
                        return [2 /*return*/, {
                                meta: {
                                    query: response.meta.query,
                                    total: response.meta.total,
                                },
                                data: mapper.map(response)
                            }];
                }
            });
        });
    };
    SearchService = __decorate([
        inversify_1.injectable()
    ], SearchService);
    return SearchService;
}(base_1.default));
exports.default = SearchService;
