// @lonelyplanet/open-planet-node/dist/resources/bookmark/bookmarkService.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = y[op[0] & 2 ? "return" : op[0] ? "throw" : "next"]) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [0, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var service_1 = require("../../service");
var bookmarkMapper_1 = require("./bookmarkMapper");
var BookmarkService = /** @class */ (function (_super) {
    __extends(BookmarkService, _super);
    function BookmarkService() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.resource = "bookmark-entry";
        _this.mapper = new bookmarkMapper_1.default();
        return _this;
    }
    BookmarkService.prototype.url = function (_a) {
        var userId = _a.userId, listId = _a.listId;
        return this.host + "/users/" + userId + "/bookmark-lists/" + listId + "/entries";
    };
    BookmarkService.prototype.addBookmarkToList = function (userId, bookmark, list) {
        return __awaiter(this, void 0, void 0, function () {
            var resource;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.http.fetch(this.url({ userId: userId, listId: list.id }), {
                            method: "POST",
                            body: {
                                data: this.mapper.toResource(bookmark),
                            },
                            headers: this.headers(),
                        })];
                    case 1:
                        resource = _a.sent();
                        return [2 /*return*/, this.mapper.map(resource)];
                }
            });
        });
    };
    BookmarkService.prototype.updateOrInsertBookmarksToList = function (userId, bookmarks, list) {
        return __awaiter(this, void 0, void 0, function () {
            var upsetUrl, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        upsetUrl = this.url({ userId: userId, listId: list.id }) + "/upsert";
                        return [4 /*yield*/, this.http.fetch(upsetUrl, {
                                method: "POST",
                                body: {
                                    data: bookmarks.map(this.mapper.toResource)
                                },
                                headers: this.headers(),
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, this.mapper.map(response)];
                }
            });
        });
    };
    BookmarkService.prototype.updateBookmark = function (userId, bookmark, list) {
        return __awaiter(this, void 0, void 0, function () {
            var resource;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.http.fetch(this.url({ userId: userId, listId: list.id }) + "/" + bookmark.id, {
                            method: "PATCH",
                            body: {
                                data: this.mapper.toResource(bookmark),
                            },
                            headers: this.headers(),
                        })];
                    case 1:
                        resource = _a.sent();
                        return [2 /*return*/, this.mapper.map(resource)];
                }
            });
        });
    };
    BookmarkService.prototype.isResourceBookmarked = function (userId, targetId) {
        return __awaiter(this, void 0, void 0, function () {
            var url, response;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        url = this.host + "/users/" + userId + "/bookmark-entries?filter[target][id][equals]=" + targetId;
                        return [4 /*yield*/, this.http.fetch(url, {
                                headers: this.headers(),
                            })];
                    case 1:
                        response = _a.sent();
                        return [2 /*return*/, this.mapper.map(response)];
                }
            });
        });
    };
    BookmarkService = __decorate([
        inversify_1.injectable()
    ], BookmarkService);
    return BookmarkService;
}(service_1.default));
exports.default = BookmarkService;
