// @lonelyplanet/open-planet-node/dist/resources/blogCategory/blogCategoryMapper.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var blogCategory_1 = require("../blogCategory");
var TYPES = require("../../types");
var mapper_1 = require("../../mapper");
var BlogCategoryMapper = /** @class */ (function (_super) {
    __extends(BlogCategoryMapper, _super);
    function BlogCategoryMapper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BlogCategoryMapper.prototype.toModel = function (resource, included) {
        if (included === void 0) { included = []; }
        var model = new blogCategory_1.default();
        model.id = resource.id;
        model.text = resource.attributes.text;
        model.slug = resource.attributes.slug;
        return model;
    };
    BlogCategoryMapper.prototype.toResource = function (model) {
        var resource = {
            id: model.id,
            type: blogCategory_1.blogCategoryType,
            attributes: {
                text: model.text,
                slug: model.slug,
            },
        };
        return resource;
    };
    __decorate([
        inversify_1.inject(TYPES.User),
        __metadata("design:type", Object)
    ], BlogCategoryMapper.prototype, "model", void 0);
    BlogCategoryMapper = __decorate([
        inversify_1.injectable()
    ], BlogCategoryMapper);
    return BlogCategoryMapper;
}(mapper_1.default));
exports.BlogCategoryMapper = BlogCategoryMapper;
exports.default = BlogCategoryMapper;
