// backpack-ui/dist/components/flag/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require("babel-runtime/helpers/extends");

var _extends3 = _interopRequireDefault(_extends2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _colors = require("../../styles/colors");

var _colors2 = _interopRequireDefault(_colors);

var _typography = require("../../utils/typography");

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

var _validReactAttributes = require("../../utils/validReactAttributes");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = (0, _extends3.default)({}, {
  backgroundColor: _colors2.default.accentGray,
  borderRadius: 2 / 8 + "em",
  color: _colors2.default.textOverlay,
  display: "inline-block",
  paddingBottom: 2 / 8 + "em",
  paddingLeft: 4 / 8 + "em",
  paddingRight: 4 / 8 + "em",
  paddingTop: 4 / 8 + "em",
  verticalAlign: "middle"
}, (0, _typography.textUppercase)(), {
  fontSize: "8px"
});

var Flag = function Flag(props) {
  var sanitizedProps = (0, _validReactAttributes.validReactAttributes)(props);

  return _react2.default.createElement(
    "span",
    (0, _extends3.default)({
      className: "Flag",
      style: [styles, props.style]
    }, sanitizedProps),
    props.children
  );
};

Flag.propTypes = {
  children: _propTypes2.default.string.isRequired,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(Flag);
