// backpack-ui/dist/components/settingBlockInput/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require("babel-runtime/helpers/extends");

var _extends3 = _interopRequireDefault(_extends2);

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _settingBlock = require("../settingBlock");

var _settingBlock2 = _interopRequireDefault(_settingBlock);

var _input = require("../form/input");

var _input2 = _interopRequireDefault(_input);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SettingBlockInput = function SettingBlockInput(props) {
  return _react2.default.createElement(
    _settingBlock2.default,
    {
      error: props.error,
      title: props.title,
      subtitle: props.subtitle,
      htmlFor: props.id
    },
    _react2.default.createElement(_input2.default, (0, _extends3.default)({}, props, {
      theme: "float",
      name: props.name,
      placeholder: props.placeholder
    }))
  );
};

SettingBlockInput.propTypes = {
  title: _propTypes2.default.string.isRequired,
  name: _propTypes2.default.string.isRequired,
  id: _propTypes2.default.string,
  placeholder: _propTypes2.default.string,
  subtitle: _propTypes2.default.string,
  error: _propTypes2.default.bool
};

exports.default = SettingBlockInput;
