// @lonelyplanet/open-planet-node/dist/resources/image/mapper.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var poi_1 = require("../poi");
var mapper_1 = require("../../mapper");
var TYPES = require("../../types");
var _1 = require("./");
var PoiMapper = /** @class */ (function (_super) {
    __extends(PoiMapper, _super);
    function PoiMapper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    PoiMapper.prototype.toModel = function (resource, included) {
        if (included === void 0) { included = []; }
        var model = new poi_1.default();
        var attrs = resource.attributes;
        model.id = resource.id;
        model.address = attrs.address;
        model.emails = attrs.emails;
        model.hours = attrs.hours_string;
        model.location = attrs.location;
        model.name = attrs.name;
        model.poiAttributes = attrs.poi_attributes;
        model.poiType = attrs.poi_type;
        model.price = attrs.price_string;
        model.publishedAt = attrs.published_at;
        model.priceRange = attrs.price_range;
        model.review = attrs.review;
        model.richHoursString = attrs.rich_hours_string;
        model.lpInternalScore = attrs.score;
        model.searchableName = attrs.searchable_name;
        model.subtypes = attrs.subtypes;
        model.telephone = attrs.telephone;
        model.telephoneInfo = attrs.telephone_info;
        model.website = attrs.website;
        model.containingPlaceId = resource.relationships["containing-place"].data.id;
        model.type = resource.type;
        var place = included.find(function (resource) { return resource.id === model.containingPlaceId; });
        if (place) {
            model.containingPlaceName = place.attributes.name;
        }
        var associations = included.filter(function (resource) {
            return resource.type === _1.imageAssociationType &&
                model.id === resource.relationships.to.data.id;
        })
            .map(function (a) { return a.relationships.from.data.id; });
        if (associations) {
            var images = included.filter(function (resource) {
                return resource.type === _1.imageType &&
                    associations.indexOf(resource.id) > -1;
            });
            model.images = images.map(function (image) { return ({
                attribution: image.attributes.attribution,
                path: image.attributes.path,
                caption: image.attributes.caption,
                height: image.attributes.height,
                width: image.attributes.width,
                orientation: image.attributes.orientation,
                tags: image.attributes.tags || [],
                url: image.links.pixels,
            }); });
        }
        return model;
    };
    PoiMapper.prototype.toResource = function (model) {
        var resource = {
            id: model.id,
            type: poi_1.poiType,
            attributes: {
                address: model.address,
                emails: model.emails,
                hours_string: model.hours,
                location: model.location,
                name: model.name,
                poi_attributes: model.poiAttributes,
                poi_type: model.poiType,
                price_range: model.priceRange,
                price_string: model.price,
                published_at: model.publishedAt,
                review: model.review,
                rich_hours_string: model.richHoursString,
                score: model.lpInternalScore,
                searchable_name: model.searchableName,
                subtypes: model.subtypes,
                telephone: model.telephone,
                telephone_info: model.telephoneInfo,
                website: model.website,
            },
        };
        return resource;
    };
    __decorate([
        inversify_1.inject(TYPES.Bookmark),
        __metadata("design:type", Object)
    ], PoiMapper.prototype, "model", void 0);
    PoiMapper = __decorate([
        inversify_1.injectable()
    ], PoiMapper);
    return PoiMapper;
}(mapper_1.default));
exports.PoiMapper = PoiMapper;
exports.default = PoiMapper;
