// backpack-ui/dist/components/settingBlockAction/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _settingBlock = require("../settingBlock");

var _moreLink = require("../moreLink/");

var _moreLink2 = _interopRequireDefault(_moreLink);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SettingBlockAction = function SettingBlockAction(_ref) {
  var description = _ref.description,
      error = _ref.error,
      title = _ref.title,
      actionText = _ref.actionText,
      subtitle = _ref.subtitle,
      onClick = _ref.onClick;
  return _react2.default.createElement(
    _settingBlock.SettingBlockButtonWrapper,
    { onClick: onClick },
    _react2.default.createElement(
      _settingBlock.SettingBlockWrapper,
      { error: error, hasAction: true },
      _react2.default.createElement(
        _settingBlock.SettingBlockButtonDescription,
        null,
        _react2.default.createElement(
          _settingBlock.SettingBlockHeader,
          { subtitle: subtitle },
          title
        ),
        description && _react2.default.createElement(
          _settingBlock.SettingBlockDescription,
          null,
          description
        )
      ),
      _react2.default.createElement(
        _moreLink2.default,
        {
          size: "small",
          style: { textAlign: "right" },
          onClick: function onClick(e) {
            return e.preventDefault();
          },
          hideIcon: true,
          isNested: true,
          caps: true
        },
        actionText
      )
    )
  );
};

SettingBlockAction.propTypes = {
  description: _propTypes2.default.string,
  actionText: _propTypes2.default.string,
  error: _propTypes2.default.bool,
  title: _propTypes2.default.string,
  subtitle: _propTypes2.default.string,
  onClick: _propTypes2.default.func
};

exports.default = (0, _radium2.default)(SettingBlockAction);
