// backpack-ui/dist/components/cardBook/menu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _socialShareContainer = require("../socialShareContainer");

var _socialShareContainer2 = _interopRequireDefault(_socialShareContainer);

var _shareMenu = require("../shareMenu");

var _shareMenu2 = _interopRequireDefault(_shareMenu);

var _iconButton = require("../iconButton");

var _iconButton2 = _interopRequireDefault(_iconButton);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var styles = {
  container: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "center"
  },

  buttonContainer: {
    marginTop: "4px"
  },

  buttonBackground: {
    borderRadius: "15% !important"
  }
};

var Menu = function Menu(_ref) {
  var downloadLink = _ref.downloadLink,
      shareText = _ref.shareText,
      shareUrl = _ref.shareUrl,
      hideShareButton = _ref.hideShareButton;
  return _react2.default.createElement(
    "div",
    { style: styles.container },
    _react2.default.createElement(_radium.Style, {
      scopeSelector: ".IconButton-background",
      rules: styles.buttonBackground
    }),
    !hideShareButton && _react2.default.createElement(
      "div",
      { style: styles.buttonContainer },
      _react2.default.createElement(
        _socialShareContainer2.default,
        {
          text: shareText,
          url: shareUrl,
          menuPosition: "left",
          hide: ["facebookMessenger", "reddit"]
        },
        _shareMenu2.default
      )
    ),
    _react2.default.createElement(
      "div",
      { style: styles.buttonContainer },
      _react2.default.createElement(_iconButton2.default, {
        shadow: true,
        iconName: "Open",
        label: "Preview and Download",
        style: [styles.button, styles.playButton],
        size: 32,
        href: downloadLink
      })
    )
  );
};

Menu.propTypes = {
  downloadLink: _propTypes2.default.string.isRequired,
  shareText: _propTypes2.default.string,
  shareUrl: _propTypes2.default.string,
  hideShareButton: _propTypes2.default.bool
};

Menu.defaultProps = {
  hideShareButton: false
};

exports.default = Menu;
