// @lonelyplanet/open-planet-node/src/openPlanetTypes.ts
export const article = "article";
export const narrative = "narrative";
export const news = "news";
export const shop = "shop";
export const topic = "topic";
export const video = "video";
export const thorntree = "thorntree";
export const thorntreeTopic = "thorntree-topic";
export const thorntreeForum = "thorntree-forum";

export { availabilityType as availability } from "./resources/availability";
export { blogCategoryType as blogCategory } from "./resources/blogCategory";
export { blogPostType as blogPost } from "./resources/blogPost";
export { bookmarkEntryType as bookmarkEntry } from "./resources/bookmark";
export { bookmarkListType as bookmarkList } from "./resources/bookmarkList";
export { bookmarkPopularityType as bookmarkPopularity } from "./resources/bookmarkPopularity";
export { imageType as image } from "./resources/image";
export { lodgingType as lodging } from "./resources/lodging";
export { partnerActivityType as partnerActivity } from "./resources/partnerActivity";
export { placeType as place } from "./resources/place";
export { poiType as poi } from "./resources/poi";
export { searchType as searchResult } from "./resources/search";
export { tagType as tag } from "./resources/tag";
export { userType as user } from "./resources/user";
export { associationType as association } from "./resources/association";
export { associationEntryType as AssociationEntry } from "./resources/associationEntry";
export { blacklistItemType as blacklistItem } from "./resources/blacklistItem";
export { tagAssociationType as tagAssociation } from "./resources/tagAssociation";
export { productType as product } from "./resources/product";
