// @lonelyplanet/open-planet-node/src/scopes.ts
const update = "u";
const read = "r";
const del = "d";
// Dotcom Connect Scopes
export const openid = "openid";
export const email = "email";
export const profile = "profile";

// Open Planet User Scopes
export const socialAccounts = "social-accounts";
export const associatedRoles = "associated-roles";
export const associatedBadges = "associated-badges";
export const userSelf = "op:users:self";
export const userSelfRead = `${userSelf}:${read}`;
export const userSelfUpdate = `${userSelf}:${update}`;
export const userSelfDelete = `${userSelf}:${del}`;
export const userSelfSocialAccountsRead = `${userSelf}:${socialAccounts}:*:${read}`;
export const userSelfSocialAccountsDelete = `${userSelf}:${socialAccounts}:*:${del}`;
export const userSelfAssociatedRolesRead = `${userSelf}:${associatedRoles}:*:${read}`;
export const userSelfAssociatedBadgesDelete = `${userSelf}:${associatedBadges}:*:${read}`;
export const userSelfemailUpdate = `${userSelf}:${email}:${update}`;

export const userSelfAll = [
  openid,
  email,
  profile,
  userSelfRead,
  userSelfUpdate,
  userSelfDelete,
  userSelfSocialAccountsRead,
  userSelfSocialAccountsDelete,
  userSelfAssociatedRolesRead,
  userSelfAssociatedBadgesDelete,
  userSelfemailUpdate,
];
