// @lonelyplanet/open-planet-node/src/types.ts
export const Auth = "Auth";
export const AuthParams = "AuthParams";
export const Availability = "Availability";
export const AvailabilityService = "AvailabilityService";
export const BlogPostMapper = "BlogPostMapper";
export const BlogPostService = "BlogPostService";
export const BlogCategoryMapper = "BlogCategoryMapper";
export const BlogCategoryService = "BlogCategoryService";
export const Bookmark = "Bookmark";
export const BookmarkList = "BookmarkList";
export const BookmarkListService = "BookmarkListService";
export const BookmarkPopularity = "BookmarkPopularity";
export const BookmarkPopularityService = "BookmarkPopularityService";
export const BookmarkService = "BookmarkService";
export const Config = "Config";
export const HttpClient = "HttpClient";
export const IClientFactory = "IClientFactory<IOpenPlanetNode>";
export const LodgingService = "LodgingService";
export const OpenPlanetHost = "OpenPlanetHost";
export const OpenPlanetNode = "OpenPlanetNode";
export const PartnerActivityService = "PartnerActivityService";
export const PlaceService = "PlaceService";
export const PoiService = "PoiService";
export const TagService = "TagService";
export const User = "User";
export const UserResourceMapper = "UserResourceMapper";
export const UserService = "UserService";
export const SearchService = "SearchService";
export const Association = "Association";
export const AssociationService = "AssociationService";
export const AssociationEntry = "AssociationEntry";
export const AssociationEntryService = "AssociationEntryService";
export const Video = "Video";
export const VideoService = "VideoService";
export const VideoSource = "VideoSource";
export const VideoSourceService = "VideoSourceService";
export const VideoPlaylist = "VideoPlaylist";
export const VideoPlaylistService = "VideoPlaylistService";
export const VideoPlaylistItem = "VideoPlaylistItem";
export const BlacklistItem = "BlacklistItem";
export const BlacklistItemService = "BlacklistItemService";
export const TagAssociation = "TagAssociation";
export const TagAssociationService = "TagAssociationService";
export const ProductService = "ProductService";
