// @lonelyplanet/dotcom-core/dist/components/autocompleteResults/autocompleteResultsList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var cn = require("classnames");

var React = require("react");

var autocompleteResultsListItem_1 = require("./autocompleteResultsListItem");

var styles = {
  "list": "lp-global-list",
  "item": "lp-global-item",
  "link": "lp-global-link",
  "image": "lp-global-image",
  "heading": "lp-global-heading",
  "description": "lp-global-description"
};

var getImageUrl = function (str) {
  return str && str.includes("imgix") ? str : "";
};

exports.AutocompleteResultsList = function (_a) {
  var items = _a.items,
      className = _a.className;
  return React.createElement("ul", {
    className: cn(styles.list, className)
  }, items.length > 0 && items.map(function (item, index) {
    var subHead = item.opType || "";
    var image = item.opType === "place" ? getImageUrl(item.thumbnail) : "";
    return React.createElement(autocompleteResultsListItem_1.AutocompleteResultsListItem, {
      title: item.title,
      key: index,
      url: item.url,
      isVideo: item.opType === "video",
      subHead: subHead,
      opType: item.opType,
      image: image
    });
  }));
};
