// @lonelyplanet/open-planet-node/dist/resources/associationEntry/mapper.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var inversify_1 = require("inversify");
var _1 = require("./");
var mapper_1 = require("../../mapper");
var mapper_2 = require("../poi/mapper");
var mapper_3 = require("../partnerActivity/mapper");
var mapper_4 = require("../place/mapper");
var mapper_5 = require("../lodging/mapper");
var mappers = {
    poi: new mapper_2.default(),
    place: new mapper_4.default(),
    lodging: new mapper_5.default(),
    partner_activity: new mapper_3.default(),
};
var AssociationEntryMapper = /** @class */ (function (_super) {
    __extends(AssociationEntryMapper, _super);
    function AssociationEntryMapper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    AssociationEntryMapper.prototype.toModel = function (resource, included) {
        if (included === void 0) { included = []; }
        var model = new _1.default();
        var attrs = resource.attributes;
        model.id = resource.id;
        model.matchesSource = attrs.matches_source;
        model.createdAt = attrs.created_at;
        model.updatedAt = attrs.updated_at;
        if (resource.relationships.target) {
            var targetResource_1 = resource.relationships.target.data;
            var target = included.find(function (i) { return i.type === targetResource_1.type && i.id === targetResource_1.id; });
            if (target) {
                var mapper = mappers[target.type];
                model.target = mapper.toModel(target, included);
            }
            else {
                model.target = resource.relationships.target.data;
            }
        }
        return model;
    };
    AssociationEntryMapper.prototype.toResource = function (model) {
        var resource = {
            id: model.id,
            type: _1.associationEntryType,
            attributes: {
                created_at: model.createdAt,
                matches_source: model.matchesSource,
                updated_at: model.updatedAt
            },
        };
        if (model.target) {
            resource.relationships = {
                target: {
                    data: {
                        type: model.target.type,
                        id: model.target.id
                    }
                }
            };
        }
        Object.keys(resource.attributes).forEach(function (key) {
            if (typeof resource.attributes[key] === "undefined") {
                delete resource.attributes[key];
            }
        });
        return resource;
    };
    AssociationEntryMapper = __decorate([
        inversify_1.injectable()
    ], AssociationEntryMapper);
    return AssociationEntryMapper;
}(mapper_1.default));
exports.AssociationEntryMapper = AssociationEntryMapper;
exports.default = AssociationEntryMapper;
