// backpack-ui/dist/components/settingBlockCheckbox/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _settingBlock = require("../settingBlock");

var _checkboxComponent = require("../checkbox/checkboxComponent");

var _checkboxComponent2 = _interopRequireDefault(_checkboxComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var SettingBlockCheckbox = function SettingBlockCheckbox(_ref) {
  var description = _ref.description,
      checked = _ref.checked,
      error = _ref.error,
      title = _ref.title,
      hasBorder = _ref.hasBorder,
      subtitle = _ref.subtitle,
      name = _ref.name,
      type = _ref.type,
      onClick = _ref.onClick,
      qaHook = _ref.qaHook;
  return _react2.default.createElement(
    _settingBlock.SettingBlockButtonWrapper,
    { onClick: onClick },
    _react2.default.createElement(
      _settingBlock.SettingBlockWrapper,
      {
        error: error,
        hasAction: true,
        hasBorder: hasBorder
      },
      _react2.default.createElement(
        _settingBlock.SettingBlockButtonDescription,
        null,
        _react2.default.createElement(
          _settingBlock.SettingBlockHeader,
          { subtitle: subtitle },
          title
        ),
        description && _react2.default.createElement(
          _settingBlock.SettingBlockDescription,
          null,
          description
        )
      ),
      _react2.default.createElement(_checkboxComponent2.default, {
        onClick: function onClick(e) {
          return e.stopPropagation();
        },
        id: "setting_" + title.toLowerCase(),
        name: name,
        checked: checked,
        size: 24,
        type: type,
        rounded: true,
        qaHook: qaHook
      })
    )
  );
};

SettingBlockCheckbox.propTypes = {
  description: _propTypes2.default.string,
  checked: _propTypes2.default.bool,
  error: _propTypes2.default.bool,
  title: _propTypes2.default.string,
  hasBorder: _propTypes2.default.bool,
  subtitle: _propTypes2.default.string,
  name: _propTypes2.default.string,
  type: _propTypes2.default.oneOf(["checkbox", "radio"]),
  onClick: _propTypes2.default.func,
  qaHook: _propTypes2.default.bool
};

SettingBlockCheckbox.defaultProps = {
  description: null,
  checked: false,
  error: false,
  title: null,
  hasBorder: false,
  subtitle: null,
  name: null,
  type: "checkbox",
  onClick: null,
  qaHook: false
};

exports.default = (0, _radium2.default)(SettingBlockCheckbox);
