// backpack-ui/dist/components/cardPrice/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _radium = require("radium");

var _radium2 = _interopRequireDefault(_radium);

var _card = require("../card");

var _price = require("../price");

var _price2 = _interopRequireDefault(_price);

var _propTypes3 = require("../../utils/propTypes");

var _propTypes4 = _interopRequireDefault(_propTypes3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var CardPrice = function CardPrice(_ref) {
  var href = _ref.href,
      imageSrc = _ref.imageSrc,
      heading = _ref.heading,
      bullets = _ref.bullets,
      price = _ref.price,
      style = _ref.style;
  return _react2.default.createElement(
    _card.Card,
    {
      className: "Card--price",
      style: [style, { minHeight: "535px" }]
    },
    _react2.default.createElement(_card.CardImage, {
      href: href,
      src: imageSrc
    }),
    _react2.default.createElement(
      _card.CardText,
      null,
      _react2.default.createElement(
        _card.CardAnchor,
        {
          href: href,
          style: { minHeight: "303px" }
        },
        bullets && bullets.length > 0 && _react2.default.createElement(_card.CardBullets, { bullets: bullets }),
        _react2.default.createElement(
          _card.CardHeading,
          null,
          heading
        ),
        price && _react2.default.createElement(_price2.default, {
          amount: price.regular,
          discountedAmount: price.sale,
          style: { bottom: "27px", position: "absolute" },
          parent: "card",
          emphasized: true
        })
      )
    )
  );
};

CardPrice.propTypes = {
  href: _propTypes2.default.string.isRequired,
  imageSrc: _propTypes2.default.string.isRequired,
  heading: _propTypes2.default.string.isRequired,
  bullets: _propTypes2.default.arrayOf(_propTypes2.default.string).isRequired,
  price: _propTypes2.default.shape({
    sale: _propTypes2.default.number,
    regular: _propTypes2.default.number
  }).isRequired,
  style: _propTypes4.default.style
};

exports.default = (0, _radium2.default)(CardPrice);
